### ISSUE TITLE
Implement DataCite usage tracker widget in the SPA UI

### ISSUE THREAD

**Issue by cmbz** [IQSS/dataverse-pm#242]
## Overview
- Implement the DataCite usage tracker widget in the SPA UI

## Dependencies
- SPA UI functionality will need to be nearing completion to perform implementation

## Related
- https://github.com/IQSS/dataverse-frontend/issues/245

### DataCite Tracker Widget Resources
- [About the DataCite Tracker Widget](https://support.datacite.org/docs/datacite-usage-tracker)
- [DataCite Usage Tracker implementation instructions](https://support.datacite.org/docs/datacite-usage-tracker#tracking-script)
- [DataCite Usage Tracker example](https://github.com/datacite/datacite-tracker/blob/main/example.html)
- [DataCite manual tracking example](https://github.com/datacite/datacite-tracker/blob/f5345d4639f2bea4ed63c912e3872884717ecd69/example.html#L21-L30)
- Note: Vivli and OSF have implemented the widget in their SPAs and experienced challenges
- [Using the DataCite MDC Tracker with Dataverse](https://docs.google.com/document/d/1cSgRVdOKkY6ouBZyqbDonZtFPNHjvBpu6DOyXHtnL8Y/edit?usp=sharing) by @qqmyers via https://github.com/IQSS/dataverse-pm/issues/4#issuecomment-1498300550 
- [#community > DataCite Usage Tracker](https://dataverse.zulipchat.com/#narrow/channel/375707-community/topic/DataCite.20Usage.20Tracker) thread on Zulip with Kelly Stathis

### Other GREI 
- https://github.com/IQSS/dataverse-pm/issues/370
- https://github.com/IQSS/dataverse-pm/issues/124
- https://github.com/IQSS/dataverse-pm/issues/196

**Comment by cmbz** [comment]
2025-04-03
- @GPortas @ekraffmiller @ChengShi-1 @g-saracca Just adding this to the Dataverse board in the frontend backlog as a reminder that we need to analyze what is needed to implement the DataCite usage tracker in the SPA.

**Comment by g-saracca** [comment]
@cmbz I've found a blocker and there is already a related open issue on their repo. See https://github.com/datacite/datacite-tracker/issues/3. We need to be able to import the `Tracker` class and `MetricType` enum from their package but it's not working.

**Comment by g-saracca** [comment]
Good point, I am just now understanding what this tool is all about. 
@cmbz If this is already done in the back-end, is it necessary to include it in the SPA?
Or does it arise as an alternative and possible replacement to using counter-processor in the back-end?

**Comment by qqmyers** [comment]
FWIW: It's an eventual replacement. It currently partially duplicates what counter-processor does (it only gets counts from the UI it's in and will miss direct API calls). But if both run at the same time, e.g. if the SPA is deployed in parallel on dataverse.harvard.edu, there would be double counts. Possibly fixable with tweaks to how the tracker or counter-processor get configured. 

For site that don't have counter-processor set up at all, the tracker may make sense now (better than nothing) - and Harvard could decide to drop the back-end counting and just lose api-only counts. So my point was just about deploying both at once on Harvard production.

**Comment by g-saracca** [comment]
ok I understand now 👍🏼 , makes sense!

**Comment by cmbz** [comment]
## Status: April 2025
- Analysis began on work needed to implement the DataCite usage tracker widget with the SPA
  - See: https://github.com/IQSS/dataverse-frontend/issues/245

**Comment by g-saracca** [comment]
@cmbz @qqmyers  I started to analyse this option since it was mentioned in this [issue](https://github.com/IQSS/dataverse-frontend/issues/245) that I am currently working on.
After some research, I understood that this widget is not exactly what the above mentioned issue needs. 
That said, I will not be implementing this widget now as we need to prioritize issue 245 for the SPA MVP release.
The good thing is that we already detected that we can't use this widget as an npm package to integrate it seamlessly into the SPA, I'll keep you posted through this issue when we take it up again.

**Comment by cmbz** [comment]
@g-saracca thanks for the update! Two follow-up questions, though.
1. Was the analysis you conducted sufficient to gain a sense of what would be needed to integrate the usage tracker into the SPA after the MVP release, by the September 2025 target date? If so, is there a summary somewhere I could look at?
2. If not, are there plans to continue with the analysis so there is a roadmap for implementation?

**Comment by g-saracca** [comment]
@cmbz no problem!

1. Not really yet. I do have the impression that it won't be something very complicated, I just need to know which events are really the ones we want to detect? views and downloads of a dataset?
2. I can continue with the analysis in this same quarter so we don't lose the momentum.

Perhaps it would be good to ask Datacite if they could provide us with a `data-repoid` identifier for testing and development purposes.

**Comment by cmbz** [comment]
Thanks @g-saracca. This is good news, and thanks for keeping the effort moving ahead. Re. DataCite credentials, @qqmyers, @jggautier, or @stevenwinship do one of you happen to know if we have the [credentials mentioned above](https://github.com/IQSS/dataverse-pm/issues/242#issuecomment-2828692234) already?

**Comment by jggautier** [comment]
Hey @cmbz and @g-saracca. I'm not sure if we do (or what a data-repoid identifier is to be honest!).

If @qqmyers or @stevenwinship don't know if we have those credentials, I could reach out to Iratxe Puebla from MDC if we think that'll help.

**Comment by qqmyers** [comment]
I don't think anyone has gotten it - it's described on https://support.datacite.org/docs/datacite-usage-tracker. Guessing you just have to ask support the same as for the JWT needed for reporting. Could probably get a data-repoid for the test DataCite account first - that could be used on beta.

That page has other useful info - they appear to be suggesting log processing in addition to the tracker to capture api calls, and they talk about what's needed to consume the info the tracker creates (since the tracker reports directly to DataCite, if we want to display those metrics, we have to go get them like we do with citations now.) I see GDCC is now maintaining counter-processor according to them too!

**Comment by cmbz** [comment]
Many thanks @jggautier and @qqmyers! 

@g-saracca looks like you just have to email support@datacite.org to request one.

**Comment by cmbz** [comment]
## Status: July 2025
- Pending

**Comment by qqmyers** [comment]
related to https://github.com/IQSS/dataverse-frontend/issues/779

**Comment by jggautier** [comment]
> This approach will also address the discrepancy between API and UI download metrics reported in this issue: https://github.com/IQSS/dataverse/issues/11623

https://github.com/IQSS/dataverse/issues/11623 is a really helpful use-case for favoring unique counts over total counts.

I think the approaches described in https://github.com/IQSS/dataverse.harvard.edu/issues/385 and https://github.com/IQSS/dataverse-frontend/issues/779 will help address but not fully address the discrepancy between API and UI download metrics reported in https://github.com/IQSS/dataverse/issues/11623, since as of Dataverse v6.9, Dataverse's UI displays total MDC views and downloads instead of unique MDC views and downloads.

There's conversation in the Dataverse Community Google Group thread at https://groups.google.com/g/dataverse-community/c/eXX1LN0rLrw about this, and I think an open and related question to address is why Dataverse displays total view and download counts instead of unique counts.

**Comment by qqmyers** [comment]
> > This approach will also address the discrepancy between API and UI download metrics reported in this issue: [IQSS/dataverse#11623](https://github.com/IQSS/dataverse/issues/11623)
> 
> [IQSS/dataverse#11623](https://github.com/IQSS/dataverse/issues/11623) is a really helpful use-case for favoring unique counts over total counts.
FWIW: Most of that issue is discussing the uniquedownloads endpoint which is a different metric that the MDC unique (total, regular, machine) counts.
> 
> I think the approaches described in [IQSS/dataverse.harvard.edu#385](https://github.com/IQSS/dataverse.harvard.edu/issues/385) and [IQSS/dataverse-frontend#779](https://github.com/IQSS/dataverse-frontend/issues/779) will help address but not fully address the discrepancy between API and UI download metrics reported in [IQSS/dataverse#11623](https://github.com/IQSS/dataverse/issues/11623), since as of Dataverse v6.9, Dataverse's UI displays total MDC views and downloads instead of unique MDC views and downloads.

"as of 6.9"? - unless I'm missing something - there was no change in what Dataverse software displays

**Comment by jggautier** [comment]
Right, by "as of" I don't mean that there was a change. I'm describing what the design decision is for the most recently released version of Dataverse.

But it doesn't hurt to make explicit that that decision hasn't changed for Dataverse versions released since we added MDC counts to the UI.
